[Bug] Regression: `SyncReturn` doesn't work for mirrored types

### Describe the bug

#1285 (since v1.79) introduces a new mechanism that replaces the old `wrap_obj` codegen-based solution to wrapping external structs. However, `wrap_sync` was not updated to make use of this, meaning that return types like `SyncReturn<Vec<Mirrored>>` won't compile.

### Codegen logs with `RUST_LOG=debug` environment variable

```shell
error[E0277]: the trait bound `Vec<polars::prelude::DataType>: flutter_rust_bridge::IntoDart` is not satisfied
   --> polars-wrapper/src/bridge.rs:627:33
    |
627 |     FLUTTER_RUST_BRIDGE_HANDLER.wrap_sync(
    |                                 ^^^^^^^^^ the trait `flutter_rust_bridge::IntoDart` is not implemented for `Vec<polars::prelude::DataType>`
    |
    = help: the following other types implement trait `flutter_rust_bridge::IntoDart`:
              Vec<T>
              Vec<chrono::DateTime<chrono::Local>>
              Vec<chrono::DateTime<chrono::Utc>>
              Vec<chrono::Duration>
              Vec<chrono::NaiveDateTime>
              Vec<f32>
              Vec<f64>
              Vec<i16>
            and 7 others
note: required by a bound in `wrap_sync`
   --> /home/vdin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/flutter_rust_bridge-1.80.1/src/handler.rs:63:18
    |
56  |     fn wrap_sync<SyncTaskFn, TaskRet>(
    |        --------- required by a bound in this associated function
...
63  |         TaskRet: IntoDart;
    |                  ^^^^^^^^ required by this bound in `Handler::wrap_sync`
```


### To Reproduce

_No response_

### Expected behavior

_No response_

### Generated binding code

_No response_

### OS

_No response_

### Version of `flutter_rust_bridge_codegen`

_No response_

### Flutter info

_No response_

### Version of `clang++`

_No response_

### Version of `ffigen`

_No response_

### Additional context

_No response_

--- a/frb_example/pure_dart/dart/lib/bridge_definitions.dart
+++ b/frb_example/pure_dart/dart/lib/bridge_definitions.dart
@@ -680,6 +680,10 @@ abstract class FlutterRustBridgeExampleSingleBlockTest {
 
   FlutterRustBridgeTaskConstMeta get kTestTuple2ConstMeta;
 
+  ApplicationSettings syncReturnMirror({dynamic hint});
+
+  FlutterRustBridgeTaskConstMeta get kSyncReturnMirrorConstMeta;
+
   Future<String> asStringMethodEvent({required Event that, dynamic hint});
 
   FlutterRustBridgeTaskConstMeta get kAsStringMethodEventConstMeta;

--- a/frb_example/pure_dart/dart/lib/bridge_generated.dart
+++ b/frb_example/pure_dart/dart/lib/bridge_generated.dart
@@ -2578,6 +2578,21 @@ class FlutterRustBridgeExampleSingleBlockTestImpl implements FlutterRustBridgeEx
         argNames: ["value"],
       );
 
+  ApplicationSettings syncReturnMirror({dynamic hint}) {
+    return _platform.executeSync(FlutterRustBridgeSyncTask(
+      callFfi: () => _platform.inner.wire_sync_return_mirror(),
+      parseSuccessData: _wire2api_application_settings,
+      constMeta: kSyncReturnMirrorConstMeta,
+      argValues: [],
+      hint: hint,
+    ));
+  }
+
+  FlutterRustBridgeTaskConstMeta get kSyncReturnMirrorConstMeta => const FlutterRustBridgeTaskConstMeta(
+        debugName: "sync_return_mirror",
+        argNames: [],
+      );
+
   Future<String> asStringMethodEvent({required Event that, dynamic hint}) {
     var arg0 = _platform.api2wire_box_autoadd_event(that);
     return _platform.executeNormal(FlutterRustBridgeTask(

--- a/frb_example/pure_dart/dart/lib/bridge_generated.io.dart
+++ b/frb_example/pure_dart/dart/lib/bridge_generated.io.dart
@@ -3821,6 +3821,14 @@ class FlutterRustBridgeExampleSingleBlockTestWire implements FlutterRustBridgeWi
   late final _wire_test_tuple_2 =
       _wire_test_tuple_2Ptr.asFunction<void Function(int, ffi.Pointer<wire_list___record__String_i32>)>();
 
+  WireSyncReturn wire_sync_return_mirror() {
+    return _wire_sync_return_mirror();
+  }
+
+  late final _wire_sync_return_mirrorPtr =
+      _lookup<ffi.NativeFunction<WireSyncReturn Function()>>('wire_sync_return_mirror');
+  late final _wire_sync_return_mirror = _wire_sync_return_mirrorPtr.asFunction<WireSyncReturn Function()>();
+
   void wire_as_string__method__Event(
     int port_,
     ffi.Pointer<wire_Event> that,

--- a/frb_example/pure_dart/dart/lib/bridge_generated.web.dart
+++ b/frb_example/pure_dart/dart/lib/bridge_generated.web.dart
@@ -1356,6 +1356,8 @@ class FlutterRustBridgeExampleSingleBlockTestWasmModule implements WasmModule {
 
   external dynamic /* void */ wire_test_tuple_2(NativePortType port_, List<dynamic> value);
 
+  external dynamic /* List<dynamic> */ wire_sync_return_mirror();
+
   external dynamic /* void */ wire_as_string__method__Event(NativePortType port_, List<dynamic> that);
 
   external dynamic /* void */ wire_sum__method__SumWith(NativePortType port_, List<dynamic> that, int y, int z);
@@ -1810,6 +1812,8 @@ class FlutterRustBridgeExampleSingleBlockTestWire
   void wire_test_tuple(NativePortType port_, List<dynamic>? value) => wasmModule.wire_test_tuple(port_, value);
 
   void wire_test_tuple_2(NativePortType port_, List<dynamic> value) => wasmModule.wire_test_tuple_2(port_, value);
+
+  dynamic /* List<dynamic> */ wire_sync_return_mirror() => wasmModule.wire_sync_return_mirror();
 
   void wire_as_string__method__Event(NativePortType port_, List<dynamic> that) =>
       wasmModule.wire_as_string__method__Event(port_, that);

--- a/frb_example/pure_dart/dart/lib/main.dart
+++ b/frb_example/pure_dart/dart/lib/main.dart
@@ -1350,6 +1350,11 @@ void main(List<String> args) async {
     expect(api.testTuple(), completion(('John', 0)));
     expect(api.testTuple(value: ('Bob', 42)), completion(('Hello Bob', 43)));
   });
+
+  test("sync return mirror", () {
+    final settings = api.syncReturnMirror();
+    testAppSettings(settings);
+  });
 }
 
 int _createGarbage() {

--- a/frb_example/pure_dart/rust/c_output_path/c_output.h
+++ b/frb_example/pure_dart/rust/c_output_path/c_output.h
@@ -825,6 +825,8 @@ void wire_test_tuple(int64_t port_, struct wire___record__String_i32 *value);
 
 void wire_test_tuple_2(int64_t port_, struct wire_list___record__String_i32 *value);
 
+WireSyncReturn wire_sync_return_mirror(void);
+
 void wire_as_string__method__Event(int64_t port_, struct wire_Event *that);
 
 void wire_sum__method__SumWith(int64_t port_, struct wire_SumWith *that, uint32_t y, uint32_t z);
@@ -1259,6 +1261,7 @@ static int64_t dummy_method_to_enforce_bundling(void) {
     dummy_var ^= ((int64_t) (void*) wire_test_struct_with_enum);
     dummy_var ^= ((int64_t) (void*) wire_test_tuple);
     dummy_var ^= ((int64_t) (void*) wire_test_tuple_2);
+    dummy_var ^= ((int64_t) (void*) wire_sync_return_mirror);
     dummy_var ^= ((int64_t) (void*) wire_as_string__method__Event);
     dummy_var ^= ((int64_t) (void*) wire_sum__method__SumWith);
     dummy_var ^= ((int64_t) (void*) wire_new__static_method__ConcatenateWith);

--- a/frb_example/pure_dart/rust/c_output_path_extra/c_output.h
+++ b/frb_example/pure_dart/rust/c_output_path_extra/c_output.h
@@ -825,6 +825,8 @@ void wire_test_tuple(int64_t port_, struct wire___record__String_i32 *value);
 
 void wire_test_tuple_2(int64_t port_, struct wire_list___record__String_i32 *value);
 
+WireSyncReturn wire_sync_return_mirror(void);
+
 void wire_as_string__method__Event(int64_t port_, struct wire_Event *that);
 
 void wire_sum__method__SumWith(int64_t port_, struct wire_SumWith *that, uint32_t y, uint32_t z);
@@ -1259,6 +1261,7 @@ static int64_t dummy_method_to_enforce_bundling(void) {
     dummy_var ^= ((int64_t) (void*) wire_test_struct_with_enum);
     dummy_var ^= ((int64_t) (void*) wire_test_tuple);
     dummy_var ^= ((int64_t) (void*) wire_test_tuple_2);
+    dummy_var ^= ((int64_t) (void*) wire_sync_return_mirror);
     dummy_var ^= ((int64_t) (void*) wire_as_string__method__Event);
     dummy_var ^= ((int64_t) (void*) wire_sum__method__SumWith);
     dummy_var ^= ((int64_t) (void*) wire_new__static_method__ConcatenateWith);

--- a/frb_example/pure_dart/rust/src/api.rs
+++ b/frb_example/pure_dart/rust/src/api.rs
@@ -1691,3 +1691,7 @@ pub fn test_tuple(value: Option<(String, i32)>) -> (String, i32) {
 pub fn test_tuple_2(value: Vec<(String, i32)>) {
     drop(value)
 }
+
+pub fn sync_return_mirror() -> SyncReturn<ApplicationSettings> {
+    SyncReturn(external_lib::get_app_settings())
+}

--- a/frb_example/pure_dart/rust/src/bridge_generated.io.rs
+++ b/frb_example/pure_dart/rust/src/bridge_generated.io.rs
@@ -858,6 +858,11 @@ pub extern "C" fn wire_test_tuple_2(port_: i64, value: *mut wire_list___record__
 }
 
 #[no_mangle]
+pub extern "C" fn wire_sync_return_mirror() -> support::WireSyncReturn {
+    wire_sync_return_mirror_impl()
+}
+
+#[no_mangle]
 pub extern "C" fn wire_as_string__method__Event(port_: i64, that: *mut wire_Event) {
     wire_as_string__method__Event_impl(port_, that)
 }

--- a/frb_example/pure_dart/rust/src/bridge_generated.rs
+++ b/frb_example/pure_dart/rust/src/bridge_generated.rs
@@ -2301,6 +2301,16 @@ fn wire_test_tuple_2_impl(
         },
     )
 }
+fn wire_sync_return_mirror_impl() -> support::WireSyncReturn {
+    FLUTTER_RUST_BRIDGE_HANDLER.wrap_sync(
+        WrapInfo {
+            debug_name: "sync_return_mirror",
+            port: None,
+            mode: FfiCallMode::Sync,
+        },
+        move || Ok(sync_return_mirror()),
+    )
+}
 fn wire_as_string__method__Event_impl(port_: MessagePort, that: impl Wire2Api<Event> + UnwindSafe) {
     FLUTTER_RUST_BRIDGE_HANDLER.wrap::<_, _, _, String>(
         WrapInfo {

--- a/frb_example/pure_dart/rust/src/bridge_generated.web.rs
+++ b/frb_example/pure_dart/rust/src/bridge_generated.web.rs
@@ -830,6 +830,11 @@ pub fn wire_test_tuple_2(port_: MessagePort, value: JsValue) {
 }
 
 #[wasm_bindgen]
+pub fn wire_sync_return_mirror() -> support::WireSyncReturn {
+    wire_sync_return_mirror_impl()
+}
+
+#[wasm_bindgen]
 pub fn wire_as_string__method__Event(port_: MessagePort, that: JsValue) {
     wire_as_string__method__Event_impl(port_, that)
 }

--- a/frb_rust/src/handler.rs
+++ b/frb_rust/src/handler.rs
@@ -53,14 +53,15 @@ pub trait Handler {
 
     /// Same as [`wrap`][Handler::wrap], but the Rust function must return a [SyncReturn] and
     /// need not implement [Send].
-    fn wrap_sync<SyncTaskFn, TaskRet>(
+    fn wrap_sync<SyncTaskFn, TaskRet, D>(
         &self,
         wrap_info: WrapInfo,
         sync_task: SyncTaskFn,
     ) -> WireSyncReturn
     where
         SyncTaskFn: FnOnce() -> Result<SyncReturn<TaskRet>> + UnwindSafe,
-        TaskRet: IntoDart;
+        TaskRet: IntoIntoDart<D>,
+        D: IntoDart;
 }
 
 /// The simple handler uses a simple thread pool to execute tasks.
@@ -120,13 +121,14 @@ impl<E: Executor, EH: ErrorHandler> Handler for SimpleHandler<E, EH> {
         });
     }
 
-    fn wrap_sync<SyncTaskFn, TaskRet>(
+    fn wrap_sync<SyncTaskFn, TaskRet, D>(
         &self,
         wrap_info: WrapInfo,
         sync_task: SyncTaskFn,
     ) -> WireSyncReturn
     where
-        TaskRet: IntoDart,
+        TaskRet: IntoIntoDart<D>,
+        D: IntoDart,
         SyncTaskFn: FnOnce() -> Result<SyncReturn<TaskRet>> + UnwindSafe,
     {
         // NOTE This extra [catch_unwind] **SHOULD** be put outside **ALL** code!
@@ -134,7 +136,7 @@ impl<E: Executor, EH: ErrorHandler> Handler for SimpleHandler<E, EH> {
         panic::catch_unwind(move || {
             let catch_unwind_result = panic::catch_unwind(move || {
                 match self.executor.execute_sync(wrap_info, sync_task) {
-                    Ok(data) => wire_sync_from_data(data.0, true),
+                    Ok(data) => wire_sync_from_data(data.0.into_into_dart(), true),
                     Err(err) => self
                         .error_handler
                         .handle_error_sync(Error::ResultError(err)),
@@ -161,14 +163,15 @@ pub trait Executor: RefUnwindSafe {
         D: IntoDart;
 
     /// Executes a Rust function that returns a [SyncReturn].
-    fn execute_sync<SyncTaskFn, TaskRet>(
+    fn execute_sync<SyncTaskFn, TaskRet, D>(
         &self,
         wrap_info: WrapInfo,
         sync_task: SyncTaskFn,
     ) -> Result<SyncReturn<TaskRet>>
     where
         SyncTaskFn: FnOnce() -> Result<SyncReturn<TaskRet>> + UnwindSafe,
-        TaskRet: IntoDart;
+        TaskRet: IntoIntoDart<D>,
+        D: IntoDart;
 }
 
 /// The default executor used.
@@ -233,14 +236,15 @@ impl<EH: ErrorHandler> Executor for ThreadPoolExecutor<EH> {
         });
     }
 
-    fn execute_sync<SyncTaskFn, TaskRet>(
+    fn execute_sync<SyncTaskFn, TaskRet, D>(
         &self,
         _wrap_info: WrapInfo,
         sync_task: SyncTaskFn,
     ) -> Result<SyncReturn<TaskRet>>
     where
         SyncTaskFn: FnOnce() -> Result<SyncReturn<TaskRet>> + UnwindSafe,
-        TaskRet: IntoDart,
+        TaskRet: IntoIntoDart<D>,
+        D: IntoDart,
     {
         sync_task()
     }

